### PR TITLE
codes for both program added

### PR DESCRIPTION
**Pull request by astikanand**

### CODE REVIEW

File: TripletFinder/tests/TestTripletFinderForGivenSum.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		@list_of_triplets.push(first_number: 4, second_number: 8, third_number: 11)
+		@list_of_triplets.push(first_number: 4, second_number: 9, third_number: 10)
+		@list_of_triplets.push(first_number: 6, second_number: 8, third_number: 9)
+		expect(@triplet_finder.getTripletForGivenSum([1, 4, 45, 6, 11, 8, 3, 19, 9, 10 ], 23)).to eq @list_of_triplets
+	end
+
+	it "should identify non-occurence of triplet" do
+		expect(@triplet_finder.getTripletForGivenSum([1, 4, 45, 6, 10, 8], 23)).to eq @list_of_triplets
+	end
+
```
Comment (ankuriitk) [inline TripletFinder/tests/TestTripletFinderForGivenSum.rb]: no negative number tests?

File: TripletFinder/tests/TestTripletFinderForGivenSum.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	}
+
+	it "should find the single occurence of triplet" do
+		@list_of_triplets.push(first_number: 4, second_number: 8, third_number: 10)
+		expect(@triplet_finder.getTripletForGivenSum([1, 4, 45, 6, 10, 8], 22)).to eq @list_of_triplets
+	end
+
+	it "should find the multiple occurence of triplet" do
+		@list_of_triplets.push(first_number: 1, second_number: 3, third_number: 19)
+		@list_of_triplets.push(first_number: 3, second_number: 9, third_number: 11)
```
Comment (ankuriitk) [inline TripletFinder/tests/TestTripletFinderForGivenSum.rb]: good!

File: FriendInviter/modules/FriendSchemaModule.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#--
+# Author: Astik Anand
+#
+# This module tells about the schema of each friend
+# in friends list json file.
+# 
+# The schema is used to check the validity of data
+#++
+module FriendSchema
+    SCHEMA = {
```
Comment (ankuriitk) [inline FriendInviter/modules/FriendSchemaModule.rb]: nice idea for having a schema for input file. However, rather than putting in code, it could be in a separate schema json file

File: FriendInviter/modules/FriendSchemaModule.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	    "type"=>"object",
+	    "required" => ["latitude", "user_id", "name", "longitude", ],
+	    "properties" => {
+	        "latitude" => {
+	            "type" => "string",
```
Comment (ankuriitk) [inline FriendInviter/modules/FriendSchemaModule.rb]: more stricter validations can be enforced in schema itelf

Comment (ankuriitk) [reply FriendInviter/modules/FriendSchemaModule.rb]: no where in the code lat/lng degree ranges are being checked

File: FriendInviter/tests/TestFriendInviter.rb [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+	end
+
+
+    ##
+    # Testcases for the Method: distanceInKmBetweenEarthCoordinates
+    #
+    # Input: longitudes and lattitudes of 2 geocoordinates
+    # Output: distance b/w 2 given geocoordinates in kms
+    # 
+    # Fact: -90 <= latitude <= 90 and -180 <= longitude <=180
```
Comment (ankuriitk) [inline FriendInviter/tests/TestFriendInviter.rb]: where is this being checked?

File: FriendInviter/tests/TestFriendInviter.rb [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    # But we don't need to verify it as they are degrees and self handled 
+    #  
+	it "should return distance in km when geoocordinates of 2 points are passed" do
+	    expect(@friend_inviter.distanceInKmBetweenEarthCoordinates(20,40,-45,220)).to eq 17235.213629906608
+	end
+
+ 
+
```
Comment (ankuriitk) [inline FriendInviter/tests/TestFriendInviter.rb]: more edge cases like north pole, equator, south pole tests

File: TripletFinder/tests/TestTripletFinderForGivenSum.rb (hunk 1 above)
Comment (astikanand) [reply TripletFinder/tests/TestTripletFinderForGivenSum.rb]: I missed that earlier and have added that and will be reflected in next commit.

File: FriendInviter/modules/FriendSchemaModule.rb (hunk 3 above)
Comment (astikanand) [reply FriendInviter/modules/FriendSchemaModule.rb]: I wanted to put in module to make use of the SCHEMA variable to use with schema validator function.

File: FriendInviter/modules/FriendSchemaModule.rb (hunk 4 above)
Comment (astikanand) [reply FriendInviter/modules/FriendSchemaModule.rb]: Actually, I got confused with an online distance calculator coz my function was giving the same value as of them for greater and lesser values. So I thought it is handled internally as it handled in other degrees problems.

But later I realized that the online portal was not giving the correct answer and hence I put a check and also wrote testcases for it.

It will be reflected in next commit.

File: FriendInviter/tests/TestFriendInviter.rb (hunk 5 above)
Comment (astikanand) [reply FriendInviter/tests/TestFriendInviter.rb]: I missed that earlier and have added that and will be reflected in next commit.

Comment (astikanand) [reply FriendInviter/tests/TestFriendInviter.rb]: I have specified the reason for not checking in earlier message.

File: FriendInviter/tests/TestFriendInviter.rb (hunk 6 above)
Comment (astikanand) [reply FriendInviter/tests/TestFriendInviter.rb]: I missed that earlier and have added that and will be reflected in next commit.

File: TripletFinder/tests/TestTripletFinderForGivenSum.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+		@list_of_triplets.push(first_number: 4, second_number: 8, third_number: 11)
+		@list_of_triplets.push(first_number: 4, second_number: 9, third_number: 10)
+		@list_of_triplets.push(first_number: 6, second_number: 8, third_number: 9)
+		expect(@triplet_finder.getTripletForGivenSum([1, 4, 45, 6, 11, 8, 3, 19, 9, 10 ], 23)).to eq @list_of_triplets
```
[the remark this answers is not in the document]
Comment (astikanand) [reply TripletFinder/tests/TestTripletFinderForGivenSum.rb]: I missed that earlier and have added that and will be reflected in next commit.
